### ISSUE TITLE
Add dynamic custom attributes to Log

### ISSUE THREAD

**Issue by mbyrne00** [newrelic/newrelic-python-agent#1275]
### Is your feature request related to a problem? Please describe.
I cannot add request-contextual data to my log entries.

### Feature Description
I want to include per-request information to each log entry so that I do not have to rely explicitly on `Transaction`. As per the [Same feature in the ruby agent](https://github.com/newrelic/newrelic-ruby-agent/issues/1141) something simple like `newrelic.agent. add_custom_log_attributes()` and `newrelic.agent. add_custom_log_attribute()` equivalents to the _existing_ `add_custom_attribute()` and `add_custom_attributes()` functions for `Transactions`. Alternatively a setting to allow transaction custom attributes to propagate to logs.

I would like to set this in my `FastAPI` `@app.middleware("http")` function much like I have been doing for `add_custom_attributes()` and hoping they would appear in log entries.

The direct contextual information is available in Splunk on legacy apps and we would like to rely less on that. 

### Alternatives
Maybe the Ruby feature isn't supporting dynamic attributes after reading this back a bit. If the logging does not manage a lifecycle like transactions do then it would be great if there was a guide for how to dynamically add attributes to certain logs with python logging and the `newrelic-admin run-program` agent running in our `Dockerfile`. is there a way to omit json logging via our app and have the outer agent understand this? Is there any other hooks? Do we have to revert to forwarding all logs manually if that's not the case and is there a guide on how to do this beginning to end with custom attributes? Or is this why Transactions exist?

I figured there must be a way if there are various contextual logging context vars optionally flushed through.

### Priority
Really want

I've searched, cloned this repo and reversed-engineered, but I'm pretty sure this feature does not exist. Very happy to be corrected if it does exist.

**Comment by KianaNZM** [comment]
Hi, Any updates on this? I wanna see my custom attributes in Log table as like Transaction table with python APM agent.

**Comment by mbyrne00** [comment]
Just feeding back that I heard from a New Relic representative and this is the information they gave me.

**TL;DR** Dynamic custom (per-http-request) attributes are not supported via the python APM agent and you need workarounds.


> The Python APM agent currently does not provide the log custom attributes function like what the Ruby APM agent supports.
> 
> However, the good news is that by default our Logs endpoint will take any JSON log message, and break that key:value pairs up into attributes. So if you write your log message in JSON format, with the extra attributes you wanted to add, then our platform will automatically split that out into an attribute associated with that log that you can then filter/query against. More info about our JSON log parsing - https://docs.newrelic.com/docs/logs/log-api/introduction-log-api/#message-attribute-parsin
>
> If for some reason you're not able to format the logs as JSON, and all the information is just ending up in the main "message" portion of the log payload, then you can leverage something called Log parsing rules in New Relic, to automatically pull out the relevant part of the log and turn it into an attribute. Here is some info about Log Parsing rules - https://docs.newrelic.com/docs/logs/ui-data/parsing/

I don't have any further information, or any plans to add this support so I'm going to close the issue and someone can raise again if need be. I suspect you'd need to back it up with a corporation that has commercial ties to New Relic to have any real sway.
